internal/venice-client-common/src/main/java/com/linkedin/venice/schema/writecompute/WriteComputeHandlerV1.java

```diff
@@ -112,6 +160,49 @@ Object updateArray(Schema arraySchema, List originalArray, Object writeComputeAr
     return originalArray;
   }
 
+  Object mergeArray(Schema arraySchema, Object currArrayUpdate, Object newArrayUpdate) {
+    if (newArrayUpdate instanceof List) {
+      return newArrayUpdate; // Partial update on a list field
+    }
+
+    List newSetUnion = (List) ((GenericRecord) newArrayUpdate).get(SET_UNION);
+    List newSetDiff = (List) ((GenericRecord) newArrayUpdate).get(SET_DIFF);
+    /**
+     * If old array update is Partial Update OP, merge it into the new collection merge OP's {@link SET_UNION} field.
+     * Otherwise, they are both collection merge OP, just merge both {@link SET_UNION} field and {@link SET_DIFF} field.
+     */
+    if (currArrayUpdate instanceof List) {
```

Comment:
If the `currArrayUpdate` is a partial update and `newArrayUpdate` is an union diff why can't we consolidate and return a partial update? Instead we are returning an union diff. That's incorrect right? What if current value for an array is [A, B, C]. A partial update is updating it to [A] and then we have an union of [B]. Instead of the expected result of partial update [A, B] we are only sending an union of [A, B]. Backend will result in [A, B, C] and not [A, B]

Reply:
Your approach can still work but it needs to generate the union diff to be union [A, B] diff [C] which is going to be more complicated (the code to figure out the diff [C] that currently is missing).

Reply:
Thanks for pointing out! Yes after offline discussion this is valid issue, I've addressed in the latest commit.